### PR TITLE
Introduce get_distrib_codename helper

### PR DESCRIPTION
**Pull request by dshcherb**
Adds a simple helper to return a distribution codename.

### CODE REVIEW

File: charmhelpers/fetch/ubuntu.py [hunk 1]
```diff
@@ -621,3 +621,7 @@ def get_upstream_version(package):
         return None
 
     return apt_pkg.upstream_version(pkg.current_ver.ver_str)
+
+
+def get_distrib_codename():
```
Comment (ajkavanagh) [inline charmhelpers/fetch/ubuntu.py]: This is logically in the wrong file.  I think it should be imported from charmhelpers.core.host like `lsb_release()` is, which, yes, does mean implementing it in the host_factory/ubuntu.py and then bubbling it through the imports.

Also, a docstring is needed.  It needs something like:

"""Return the codename of the distribution

:returns: The codename
:rtype: str
"""

**Comment by ajkavanagh** [comment]
@dshcherb (and reminder to merger or self): it ought to have a 'Closes-Bug: #254' in the commit message when its merged to auto-close the bug.

**Comment by dshcherb** [comment]
Pull-request updated, HEAD is now fba1b133cea6b0e29f8e438058a0f8da25b98d1d

**Comment by ajkavanagh** [comment]
Yeah, except it didn't deal with the issue that the rest of the code in the module hasn't been refactored to use the function; a bit of a negative and why I was still -ve on my review ...

**Comment by dshcherb** [comment]
@ajkavanagh

I addressed that in the review about the ubuntu.py rework for gpg and proxy handling. Although this seems to be the right place to do it

**Comment by ajkavanagh** [comment]
@dshcherb could you link to that review from here please.

**Comment by dshcherb** [comment]
https://github.com/juju/charm-helpers/pull/248

I can rebase it on top of this change now (I merged this branch before it was merged in that PR).

**Pull request merged**